include/numerics/petsc_vector.h

```diff
@@ -533,14 +546,33 @@ PetscVector<T>::PetscVector (const Parallel::Communicator & comm_in,
   _array_is_present(false),
   _local_form(nullptr),
   _values(nullptr),
-  _global_to_local_map(),
+  _global_to_local_map(nullptr),
```

Comment:
I feel like I just concluded, when talking to @jwpeterson a few weeks back, that we didn't have many places in the library where moving to C++ delegating constructors would be useful ... and yet here we are.  Now I wish I could remember *why* I'd come to that conclusion.

Reply:
Not sure exactly what you are asking, but AFAIK this line doesn't change anything and could be omitted if desired, as it's just calling the default `shared_ptr` constructor.

Reply:
`_global_to_local_map` will be initialized regardless?  What I want is that if check `_global_to_local_map`, it should return `0`

Reply:
Oh, not asking about this line in this spot in particular.  Just noticing the same line being added again, and again, and again, in different constructors that I'd naively think could all be delegating most of their work to a single target constructor, then handling only their minor differences themselves.